backend/cn/lib/solver.ml

```diff
@@ -1259,17 +1268,17 @@ let model_evaluator solver mo =
         globals = gs
       }
     in
-    declare_solver_basics evaluator;
-    List.iter (debug_ack_command evaluator) defs;
+    if new_solver then declare_solver_basics evaluator;
     fun e ->
       push evaluator;
+      List.iter (debug_ack_command evaluator) defs;
```

Comment:
Ah yes, good call. I added a guard for this.

Reply:
Hmm yeah, good point. We cannot load the definitions when the solver is created anymore because it's only created once, so we need to change the push/pop behavior to only pop when the requested model is different from the one we currently have. Will look into this.